include/chaiscript/language/chaiscript_engine.hpp

```diff
@@ -505,7 +505,7 @@ namespace chaiscript
 
         // Let's see if this is a link that we should expand
         std::vector<char> buf(2048);
-        const size_t pathlen = readlink(dllpath.c_str(), &buf.front(), buf.size());
+        const size_t pathlen = static_cast<size_t>(readlink(dllpath.c_str(), &buf.front(), buf.size()));
```

Comment:
Note that just `static_cast<size_t>` here fails to address the problem that readlink can actually return a negative value, which I look for on the very next line. So there's been a lingering bug here for quite some time.